### PR TITLE
explain fields in `defaultContext`

### PR DESCRIPTION
**Pull request by tanyabouman**
Hakyll.Web.Template.Context
- add examples to the fields
- add explanation about replacing `defaultContext` fields

'More on compilers: load, and templates' tutorial
- add `$title$` to the list of fields in `defaultContext`
- add explanation about replacing `defaultContext` fields

### CODE REVIEW

**Comment by tanyabouman** [comment]
I used a commit from #1025 to get CI to pass.

File: lib/Hakyll/Web/Template/Context.hs [hunk 1]
```diff
@@ -288,7 +299,16 @@ metadataField = Context $ \k _ i -> do
...
+-- | Absolute url to the resulting item.  For an example
+-- underlying file @posts/foo.markdown@ and result file @posts/foo.html@,
+-- the context
+--
+-- > urlField "url"
+--
+-- is equivalent to
+--
+-- > constField "url" "posts/foo.html"
+--
```
Comment (Minoru) [inline lib/Hakyll/Web/Template/Context.hs:311]: I find this example a bit contrived. Can't it say that "for an item that produces a file _posts/foo.html_, this field contains "posts/foo.html"?

Ditto for `pathField` and `titleField`; I feel those examples would be more readable as prose, not code.

**Review by Minoru — COMMENTED** [review]
I like this overall, but I've got a suggestion about `constField` in examples.

File: lib/Hakyll/Web/Template/Context.hs (hunk 1 above)
Comment (tanyabouman) [reply lib/Hakyll/Web/Template/Context.hs]: I switched the examples to prose.  What do you think?

File: web/tutorials/04-compilers.markdown [hunk 2]
```diff
@@ -114,8 +114,22 @@ you to use:
 - `$body$` for the body of the page;
 - `$url$` for the destination URL of the page;
 - `$path$` for the original filepath of the page;
+- `$title` for the basename of the original filepath of the page;
```
Comment (Minoru) [inline web/tutorials/04-compilers.markdown:117]: A typo (missed it yesterday):

```suggestion
- `$title$` for the basename of the original filepath of the page;
```

File: lib/Hakyll/Web/Template/Context.hs (hunk 1 above)
Comment (Minoru) [reply lib/Hakyll/Web/Template/Context.hs]: I think it's perfect now, thank you!

**Review by Minoru — APPROVED** [review]

**Pull request merged**